src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -69,97 +234,135 @@ public void runCargo(boolean a,boolean reversed) {
     double[] returnArray = new double[2];
-    returnArray[0] = UnitConversion(velocity, Constants.gearDiameter); //Converts FPS to RPM
+    returnArray[0] = UnitConversion(velocity, Constants.Shooter.gearDiameter); //Converts FPS to RPM
     returnArray[1] = hoodAngle;
     return returnArray;
-
   }
-
   public double UnitConversion(double KBallSpeed, double GearDiameter) {
     // Convert from FPS of the ball into RPM
-    return (((KBallSpeed * 12) / Constants.gearDiameter) * Constants.ballFlywheelratio) * 2;
+    return (((KBallSpeed * 12) / Constants.Shooter.gearDiameter) * Constants.Shooter.ballFlywheelratio) * 2;
   }
 
+
+
+
   public void prime() {
+     int aimMode = (int)DShootingMode.getDouble(0);
     // Check what aimMode is active, gets distance if AUTO, winds flywheel, adjusts
     // hood correspondingly
+    
     switch (aimMode) {
       case 0: // Case for LOW mode, winds flywheel to preset RPM and adjusts hood to preset
               // angle
-        adjustHood(Constants.LOWAngle);
-        windFlywheel(Constants.LOWRPM);
+        adjustHood(Constants.Shooter.LOWAngle);
+        windFlywheel(Constants.Shooter.LOWRPM);
         break;
       case 1: // Case for AUTO mode, calculates trajectory and winds flywheel/adjusts hood to
               // a dynamic state
-        adjustHood(ProjectilePrediction(Constants.shooterHeight, 0, Constants.highHeight, getDistance(),
-            Constants.gravity, Constants.airboneTime)[1]);
-
-        windFlywheel((int) (Math.ceil(ProjectilePrediction(Constants.shooterHeight, 0, Constants.highHeight,
-            getDistance(), 32, Constants.airboneTime)[0])));
+              automode();
+       
 
         break;
       case 2: // Case for LAUNCH mode, winds flywheel to preset RPM and adjusts hood to preset
               // angle
-        adjustHood(Constants.LAUNCHAngle);
-        windFlywheel(Constants.LAUNCHRPM);
+        adjustHood(Constants.Shooter.LAUNCHAngle);
+        windFlywheel(Constants.Shooter.LAUNCHRPM);
         break;
       case 3: // Case for TARMAC mode, winds flywheel to preset RPM and adjusts hood to preset
               // angle
-        adjustHood(Constants.TARMACAngle);
-        windFlywheel(Constants.TARMACRPM);
+        adjustHood(Constants.Shooter.TARMACAngle);
+        windFlywheel(Constants.Shooter.TARMACRPM);
         break;
       case 4: //Case for TEST mode, just takes an RPM and winds
-        windFlywheel(3700);
+        windFlywheel(DShooterRPMInput.getDouble(0));
         break;
     }
+    
   }
 
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
-    SmartDashboard.putNumber("dist", getDistance());
-    SmartDashboard.putNumber("RPM", KShooterEncoder.getVelocity());
+    //SmartDashboard.putNumber("IAccum",KShooterController.getIAccum());
+    //SmartDashboard.putNumber("dist", getDistance());
+    IDelayTable.setNumber(idelay);
```

Comment:
idelay is a loop variable?  this feels like a test thing, and we should remove it.

Reply:
talked about this with Jim, addressing in a next pull request.